pinot-core/src/test/java/org/apache/pinot/queries/SelectionOnlyEarlyTerminationTest.java

```diff
@@ -0,0 +1,85 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.queries;
+
+import org.apache.pinot.common.response.broker.BrokerResponseNative;
+import org.apache.pinot.core.operator.CombineOperator;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+
+/**
+ * Tests selection only queries
+ */
+public class SelectionOnlyEarlyTerminationTest extends BaseSingleValueQueriesTest {
+
+  /**
+   * In order to ensure each thread is executing more than 1 segment, this test is against
+   * (2 * MAX_NUM_THREADS_PER_QUERY) segments.
+   */
+  @Override
+  protected int getNumSegmentDataManagers() {
+    return CombineOperator.MAX_NUM_THREADS_PER_QUERY * 2;
+  }
+
+  /**
+   * With early termination, Selection Only query is scheduled with getNumSegmentDataManagers() threads,
+   * total segment processed is same at num threads.
+   */
+  @Test
+  public void testSelectOnlyQuery() {
+    int numThreadsPerQuery = getNumSegmentDataManagers();
```

Comment:
Why `numThreadsPerQuery` != `CombineOperator.MAX_NUM_THREADS_PER_QUERY`?

Reply:
Cause this test will put instance responses twice in dataTableMap for broker reduce, I will update the parameters to reflect this setup.

Reply:
updated the PR